addons/binding/org.openhab.binding.sinope/src/main/java/org/openhab/binding/sinope/internal/discovery/SinopeThingsDiscoveryService.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.sinope.internal.discovery;
+
+import java.io.IOException;
+import java.util.Set;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryListener;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.sinope.SinopeBindingConstants;
+import org.openhab.binding.sinope.handler.SinopeGatewayHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.tulip.sinope.util.ByteUtil;
+
+/**
+ * The {@link AbstractDiscoveryService} provides methods which handle the {@link DiscoveryListener}s.
+ *
+ * Subclasses do not have to care about adding and removing those listeners.
+ * They can use the protected methods {@link #thingDiscovered(DiscoveryResult)} and {@link #thingRemoved(String)} in
+ * order to notify the registered {@link DiscoveryListener}s.
+ *
+ * @author Pascal Larin - Initial contribution
+ *
+ */
+public class SinopeThingsDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(SinopeThingsDiscoveryService.class);
+
+    private final static int SEARCH_TIME = 120;
+
+    private SinopeGatewayHandler sinopeGatewayHandler;
+
+    public SinopeThingsDiscoveryService(SinopeGatewayHandler sinopeGatewayHandler) {
+        super(SEARCH_TIME);
+        this.sinopeGatewayHandler = sinopeGatewayHandler;
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypes() {
+        return SinopeBindingConstants.SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public void startScan() {
+        logger.info("Sinope Things starting scan");
+        try {
+            sinopeGatewayHandler.startSearch(this);
+        } catch (IOException e) {
+            logger.error("Search failed with an exception", e);
```

Comment:
See above, a user cannot deal with such a message.

Reply:
Updated to debug